pkg/apiserver/registry/system/supportbundle/rest_test.go

```diff
@@ -174,3 +181,110 @@ func TestControllerStorage(t *testing.T) {
 		Status:     system.SupportBundleStatusNone,
 	}, object)
 }
+
+func TestNewAgentStorage(t *testing.T) {
+	defaultFS = afero.NewMemMapFs()
+	defaultExecutor = new(testExec)
+	defer func() {
+		defaultFS = afero.NewOsFs()
+		defaultExecutor = exec.New()
+	}()
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	fakeOVSCtl := ovsctltest.NewMockOVSCtlClient(ctrl)
+	fakeAgentQuerier := agentqueriertest.NewMockAgentQuerier(ctrl)
+	fakeNetworkPolicyQuerier := queriertest.NewMockAgentNetworkPolicyInfoQuerier(ctrl)
+	fakeIPTables := iptablestest.NewMockInterface(ctrl)
+	fakeOVSCtl.EXPECT().DumpFlows()
+	fakeOVSCtl.EXPECT().DumpPortsDesc()
+	fakeNetworkPolicyQuerier.EXPECT().GetAddressGroups()
+	fakeNetworkPolicyQuerier.EXPECT().GetAppliedToGroups()
+	fakeNetworkPolicyQuerier.EXPECT().GetNetworkPolicies(&querier.NetworkPolicyQueryFilter{})
+	fakeAgentQuerier.EXPECT().GetAgentInfo(&clusterinformationv1beta1.AntreaAgentInfo{}, false)
+	fakeIPTables.EXPECT().Save()
+	storage := NewAgentStorage(fakeOVSCtl, fakeAgentQuerier, fakeNetworkPolicyQuerier, fakeIPTables)
+	_, err := storage.SupportBundle.Create(context.TODO(), &system.SupportBundle{
```

Comment:
nit: I feel like it makes a bit more sense for unit test to define `ctx := context.Background()` at the beginning of the test case and use that context for every function that requires it. I feel like `Background` is more logical than `TODO` for tests, given that we don't intend to ever replace this `context.TODO()` with something else.

Reply:
Some tests for rest API customize context to set namespace:
https://github.com/antrea-io/antrea/blob/5e96c8c2e66a67b10efc03031639c817639b6960/pkg/apiserver/registry/stats/networkpolicystats/rest_test.go#L292

Some tests customize context to get a chance to cancel it:
https://github.com/antrea-io/antrea/blob/5e96c8c2e66a67b10efc03031639c817639b6960/pkg/apiserver/registry/system/supportbundle/rest_test.go#L70

You mean creating a global background and use it for all tests, and create child context from it if needed?

Reply:
not global, but for a given test case if that test case requires a context object:
```go
func MyTest(t *testing.T) {
        ctx := context.Bakground()
        obj1.req(ctx, ...)
        obj2.req(ctx, ...)
        ctx, cancelFunc := context.WithCancel(ctx)
        defer cancelFunc()
        obj3.req(ctx, ...)
}
```

Reply:
updated, thanks